Objects/Converters/ConverterAutocadCivil/ConverterAutocadCivilShared/ConverterAutocadCivil.Civil.cs

```diff
@@ -1065,8 +1065,23 @@ private List<CivilDataField> PartDataRecordToSpeckle(PressureNetworkPartData par
   // TODO: add pressure fittings
   public Pipe PipeToSpeckle(CivilDB.Pipe pipe)
   {
-    ICurve curve = CurveToSpeckle(pipe.BaseCurve);
-
+    // get the pipe curve
+    // rant: if this is a straight or curved pipe, the BaseCurve prop is fake news && will return a DB.line with start and endpoints set to [0,0,0] & [0,0,1]
+    // do not use CurveToSpeckle(basecurve) 😡
+    ICurve curve;
+    switch (pipe.SubEntityType)
+    {
+      case PipeSubEntityType.Straight:
+        var line = new Acad.LineSegment3d(pipe.StartPoint, pipe.EndPoint);
+        curve = LineToSpeckle(line);
+        break;
+      case PipeSubEntityType.Curved:
+        curve = ArcToSpeckle(pipe.Curve2d);
+        break;
+      default:
+        curve = CurveToSpeckle(pipe.BaseCurve); // basecurve is fake news, but we're still sending the other types with props for now
```

Comment:
Should we send it?

Reply:
I'm saying yes since previously we were: the meshes and other props on these pipes should still be correct, just not the basecurve